databuilder/tables/beta/tpp.py

```diff
@@ -87,6 +105,46 @@ class clinical_events(EventFrame):
     ctv3_code = Series(CTV3Code)
     numeric_value = Series(float)
 
+    def with_codes(
+        self,
+        codelist,
+        after=None,
+        before=None,
+        on_or_after=None,
+        on_or_before=None,
+        between=None,
+        on_or_between=None,
+    ):
```

Comment:
Would we want to provide some validation that users weren't providing eg both `before` and `on_or_before`, or `before` and `between`?